third_party/lua-yaml/lyaml.cc

```diff
@@ -610,6 +611,44 @@ static int yaml_is_flow_mode(struct lua_yaml_dumper *dumper) {
    return 0;
 }
 
+/**
+ * Copy @a in_str to @a out_str with illegal UTF-8 sequences replaced by U+FFFD.
+ * Return length of output string.
+ */
+static size_t
+filter_invalid_utf8(const uint8_t *in_str, ssize_t in_len,
+                    uint8_t *out_str, ssize_t out_size)
+{
+   int in_offset = 0;
+   int out_offset = 0;
+   int out_capacity = out_size - 1;
+   UBool is_error = false;
+
+   while (in_offset < in_len && out_capacity > 0 && !is_error) {
+      UChar32 c;
+      U8_NEXT(in_str, in_offset, in_len, c);
+      if (c == U_SENTINEL)
+         c = 0xfffd;
+      U8_APPEND(out_str, out_offset, out_capacity, c, is_error);
+      out_capacity = out_size - out_offset - 1;
+   }
+
+   out_str[out_offset] = '\0';
+   return out_offset;
+}
+
+/**
+ * Dump the error message from @a err to the output buffer @a buf.
+ * Return length of output string.
+ */
+static size_t
+error_to_string(const struct error *err, char *buf, size_t buf_size)
+{
+   const char *msg = err->errmsg;
+   return filter_invalid_utf8((const uint8_t *)msg, strlen(msg),
+                              (uint8_t *)buf, buf_size);
```

Comment:
Why bother formatting a malformed error message? Why not just use `utf8_check_printable` and encode the message in base64 if it fails, like we do with malformed strings? After all, a malformed error message means that something is wrong. Having a base64 output is more useful, because by looking at it the user can figure out what's broken.

Reply:
Done